jormungandr/src/blockchain/process.rs

```diff
@@ -312,22 +322,29 @@ async fn process_and_propagate_new_ref(
     mut network_msg_box: MessageBox<NetworkMsg>,
 ) -> chain::Result<()> {
     let header = new_block_ref.header().clone();
-    tracing::debug!("processing the new block and propagating");
-    // Even if this fails because the queue is full we periodically recompute the tip
-    tip_update_mbox
-        .try_send(new_block_ref)
-        .unwrap_or_else(|err| {
-            tracing::error!(
-                "cannot send new ref to be evaluated as candidate tip: {}",
-                err
-            )
-        });
+    let span = span!(Level::DEBUG, "process_and_propagate_new_ref", block = %header.hash());
+
+    async {
```

Comment:
Can we use `span.enter()` here as well? Using `.instrument()` doesn't really look good in most cases and adds an unnecessary indentation level.

Reply:
holding the guard returned by `span.enter()` across await points (line 341) will result in incorrect traces. If we want to log that function inside of a span this is the only way.

I also think that `span.enter()` should be avoided if possible in async functions, because it may be ok when we first introduce it, but successive modifications could introduce await points after it and not realize that spans are now screwed up (as no errors are returned by the compiler).